src/fastga_he/gui/power_train_network_viewer.py

```diff
@@ -57,65 +187,437 @@ def power_train_network_viewer(
         names,
         connections,
         components_type,
+        components_om_type,
         icons_name,
         icons_size,
     ) = configurator.get_network_elements_list()
-    distance_from_prop_loads, prop_loads = configurator.get_distance_from_propulsive_load()
 
-    for component_name, component_type, icon_name, icon_size in zip(
-        names, components_type, icons_name, icons_size
+    # Build node attributes dictionaries
+    node_sizes = {}
+    node_types = {}
+    node_om_types = {}
+    node_icons = {}
+
+    # For animation purposes
+    node_image_sequences = {}
+    propeller_rotation_sequences = {}
+
+    for component_name, component_type, om_type, icon_name, icon_size in zip(
```

Comment:
Is there no hierarchy to the placement of the node ? Does PyGraphviz handle it on its own ?